mas-foundation/src/main/java/com/ca/mas/core/policy/PolicyManager.java

```diff
@@ -20,28 +20,67 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import static com.ca.mas.foundation.MASConstants.LOGOUT_EXTRA;
+
 /**
  * Keeps track of policies that can be applied to requests.
  */
 public class PolicyManager {
 
     private final MssoContext mssoContext;
-    private final List<MssoAssertion> defaultPolicy = new ArrayList<>();
     private final Object policySync = new Object();
+    private final StorageReadyAssertion storageReadyAssertion;
+    private final SecureLockAssertion secureLockAssertion;
+    private ClientCredentialAssertion clientCredentialAssertion;
+    private DeviceRegistrationAssertion deviceRegistrationAssertion;
+    private AccessTokenAssertion accessTokenAssertion;
+    private LocationAssertion locationAssertion;
+    private TelephoneAssertion telephoneAssertion;
+    private CustomHeaderAssertion customHeaderAssertion;
+    private ResponseRecoveryAssertion responseRecoveryAssertion;
 
     public PolicyManager(MssoContext mssoContext) {
         this.mssoContext = mssoContext;
+        storageReadyAssertion = new StorageReadyAssertion();
+        secureLockAssertion = new SecureLockAssertion();
+        clientCredentialAssertion = new ClientCredentialAssertion();
+        deviceRegistrationAssertion = new DeviceRegistrationAssertion();
+        accessTokenAssertion = new AccessTokenAssertion();
+        locationAssertion = new LocationAssertion();
+        telephoneAssertion = new TelephoneAssertion();
+        customHeaderAssertion = new CustomHeaderAssertion();
+        responseRecoveryAssertion = new ResponseRecoveryAssertion();
+    }
+
+    // - By default use true
+    private List<MssoAssertion> getSwapPolicy(boolean useDefaultPolicy) {
```

Comment:
1. I would suggest building 2 Policies (default, logout) when constructing the PolicyManger, in that case, we don't have to create the list everything we invoke the policy.
2. May create a policies map with policy name as the key Map<String, List<MssoAssertion>> policies. In this case we can extend and support more policy in the future. The policy name would be the url of the request, find the policy from the policies map using the request url, if found use that policy, if not found execute the default policy.